health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/MainActivity.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.exercisesamplecompose
+
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.fragment.app.FragmentActivity
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.NavHostController
+import androidx.wear.compose.navigation.rememberSwipeDismissableNavController
+import com.example.exercisesamplecompose.data.HealthServicesManager
+import com.example.exercisesamplecompose.presentation.ExerciseSampleApp
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class MainActivity : FragmentActivity() {
+
+    private lateinit var navController: NavHostController
+
+    @Inject
+    lateinit var healthServicesManager: HealthServicesManager
```

Comment:
I think you probably don't need this injected here if you're able to use hiltViewModel (See other comment) and you can therefore keep HealthServicesManager out of any composable.

Reply:
This was giving me grief for some reason, kept breaking it when I removed the Inject. Is this a blocker?

Reply:
I don't think so. I think my thought was that you can completely move any dependence on HealthServicesManager in the Activity, and so not pass it to ExerciseSampleApp as it doesn't need it: In ExerciseSampleApp, you can then use val viewModel = hiltViewModel<ExerciseViewModel>() and have HealthServicesManager injected in the ViewModel.

I don't think it's a blocker, or necessarily right, but it does mean that your compose code has no dependencies on the HealthServicesManager, which I think it desireable.